InstantSearch.js/segment-connector/src/app.js

```diff
@@ -0,0 +1,127 @@
+/* global analytics */
+
+import instantsearch from 'instantsearch.js';
+import {
+  searchBox,
+  hits,
+  refinementList,
+  pagination,
+  rangeSlider,
+} from 'instantsearch.js/es/widgets';
+import { createInsightsMiddleware } from 'instantsearch.js/es/middlewares';
+import algoliasearch from 'algoliasearch/lite';
+
+const searchClient = algoliasearch(
+  'latency',
+  '6be0576ff61c053d5f9a3225e2a90f76'
+);
+
+const search = instantsearch({
+  indexName: 'instant_search',
+  searchClient,
+});
+
+search.addWidgets([
+  searchBox({
+    container: '#searchbox',
+  }),
+  hits({
+    container: '#hits',
+    templates: {
+      item: (hit, bindEvent) => {
+        const result = `
+        <article>
+          <h1>${instantsearch.highlight({ attribute: 'name', hit })}</h1>
+          <p>${instantsearch.highlight({ attribute: 'description', hit })}</p>
+          <button type="button" ${bindEvent('view', hit, 'View')}>
+            View Product
+          </button>
+          <button type="button" ${bindEvent('click', hit, 'Favorite')}>
+            Favorite
+          </button>
+          <button type="button" ${bindEvent('conversion', hit, 'Add to Cart')}>
+            Add to Cart
+          </button>
+          <button type="button" ${bindEvent('conversion', hit, 'Order')}>
+            Order Now
+          </button>
+        </article>
+      `;
+        return result;
+      },
+    },
+  }),
+  refinementList({
+    container: '#brand-list',
+    attribute: 'brand',
+  }),
+  rangeSlider({
+    container: '#price',
+    attribute: 'price',
+  }),
+  pagination({
+    container: '#pagination',
+  }),
+]);
+
+// Set userToken to Segment
+analytics.identify('my-user-token');
+
+const insightsMiddleware = createInsightsMiddleware({
+  insightsClient: null,
+  onEvent(event) {
+    const { widgetType, eventType, payload, hits, attribute } = event;
+
+    console.log(event);
```

Comment:
remove the log?

Reply:
I intentionally left it so that they see what's going on as soon as they run the example, but you think too much?

Reply:
I think people will probably leave this log in their production sites, but that's probably not a big problem